tests/components/smlight/test_config_flow.py

```diff
@@ -0,0 +1,425 @@
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result.get("type") == FlowResultType.FORM
+    assert result.get("step_id") == "user"
+
+
+async def test_user_device_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort user flow if device already configured."""
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=False,
+        ),
+    ):
+        mock_config_entry.add_to_hass(hass)
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={
+                CONF_HOST: "slzb-06.lan",
+            },
+        )
+
+        assert result.get("type") is FlowResultType.ABORT
+        assert result.get("reason") == "already_configured"
+
+
+async def test_zeroconf_device_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort zeroconf flow if device already configured."""
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=False,
+        ),
+    ):
+        mock_config_entry.add_to_hass(hass)
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+            data=DISCOVERY_INFO,
+        )
+
+        assert result.get("type") is FlowResultType.ABORT
+        assert result.get("reason") == "already_configured"
+
+
+async def test_user_invalid_auth(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we handle invalid auth."""
+    with (
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=True,
+        ),
+        patch(
+            "pysmlight.web.Api2.authenticate",
+            side_effect=SmlightAuthError,
+        ),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={
+                CONF_HOST: "slzb-06.local",
+            },
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result.get("step_id") == "auth"
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_USERNAME: "test",
+                CONF_PASSWORD: "bad",
+            },
+        )
+
+        assert result2["type"] == FlowResultType.FORM
```

Comment:
always finish the tests to terminal states, to show we can recover (create entry/abort) by fixing the error and continuing

Reply:
I have fixed this one to recover with correct credentials, will see which others it makes sense for, some of the errors are really fatal though.